stock_vertical_lift/models/vertical_lift_shuttle.py

```diff
@@ -0,0 +1,276 @@
+        action = self.env.ref(action_xmlid).read()[0]
+        action["target"] = "main"
+        return action
+
+    def action_manual_barcode(self):
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": "vertical.lift.shuttle.manual.barcode",
+            "view_mode": "form",
+            "name": _("Barcode"),
+            "target": "new",
+        }
+
+    # TODO: should the mode be changed on all the shuttles at the same time?
+    def switch_pick(self):
+        self.mode = "pick"
+        self.release_vertical_lift_tray()
+        return self.action_open_screen()
+
+    def switch_put(self):
+        self.mode = "put"
+        self.release_vertical_lift_tray()
+        return self.action_open_screen()
+
+    def switch_inventory(self):
+        self.mode = "inventory"
+        self.release_vertical_lift_tray()
+        return self.action_open_screen()
+
+    def _hardware_vertical_lift_release_tray_payload(self):
+        """Prepare "release" message to be sent to the vertical lift hardware
+
+        Private method, this is where the implementation actually happens.
+        Addons can add their instructions based on the hardware used for
+        this location.
+
+        The hardware used for a location can be found in:
+
+        ``self.hardware``
+
+        Each addon can implement its own mechanism depending of this value
+        and must call ``super``.
+
+        The method must send the command to the vertical lift to release (close)
+        the tray.
+
+        Returns a message in bytes, that will be sent through
+        ``VerticalLiftShuttle._hardware_send_message()``.
+        """
+        if self.hardware == "simulation":
+            message = _("Releasing tray")
+            return message.encode("utf-8")
+        else:
+            raise NotImplementedError()
+
+    def release_vertical_lift_tray(self):
+        """Send instructions to the vertical lift hardware to close trays
+
+        The actual implementation of the method goes in the private method
+        ``_hardware_vertical_lift_release_tray()``.
+        """
+        self.ensure_one()
+        payload = self._hardware_vertical_lift_release_tray_payload()
+        return self._hardware_send_message(payload)
+
+    def _send_notification_refresh(self, success):
+        """Send a refresh notification to the current opened screen
+
+        The form controller on the front-end side will instantaneously
+        refresh the form with the latest committed data.
+
+        It can be used for instance after a vertical lift hardware
+        event occurred to inform the user on their screen.
+
+        The method is private only to prevent xml/rpc calls to
+        interact with the screen.
+        """
+        # XXX do we want to do something special in the notification?
+        self._operation_for_mode()._send_notification_refresh()
+
+
+class VerticalLiftShuttleManualBarcode(models.TransientModel):
```

Comment:
Could you maybe split one model per file ?

Reply:
ping @dzungtran89

Reply:
Good day @rousseldenis, @mmequignon, the model has been split to https://github.com/OCA/stock-logistics-warehouse/blob/8ba7d977a5f513f934c6d23ce47f59a304972a5c/stock_vertical_lift/wizards/vertical_lift_shuttle_manual_barcode.py. This PR has been rebased and updated with latest test requirement. Thanks for your comments.

Reply:
I am checking the travis...